test/app-db/src/main/java/org/glassfish/soteria/test/ApplicationConfig.java

```diff
@@ -42,12 +42,13 @@
 import javax.enterprise.context.ApplicationScoped;
 import javax.inject.Named;
 import javax.security.enterprise.identitystore.DatabaseIdentityStoreDefinition;
+import javax.security.enterprise.identitystore.Pbkdf2PasswordHash;
 
 @DatabaseIdentityStoreDefinition(
     dataSourceLookup="${'java:global/MyDS'}", 
     callerQuery="#{'select password from caller where name = ?'}",
     groupsQuery="select group_name from caller_groups where caller_name = ?",
-    hashAlgorithm = TestPlaintextPasswordHash.class,
+    hashAlgorithm = Pbkdf2PasswordHash.class,
     hashAlgorithmParameters = {
```

Comment:
You are correct. I've manually deployed the WAR on my machine and tried to access the page and I get a 500 error because of that exception. The test is now not passing on my machine. I'll have to check how is Travis accepting it.

But anyway, I don't think invalid parameters should cause an error but instead it should just be ignored. That's the way it works on JPA and it's very helpful to support vendor specific options that have different names on each implementation.

The JPA spec even says explicitly that unrecognized options must be ignored: http://docs.oracle.com/javaee/7/api/javax/persistence/EntityManager.html#setProperty-java.lang.String-java.lang.Object-